frontend/src/components/login/Login.js

```diff
@@ -23,27 +21,43 @@ class Login extends Component {
     let userData = this.state;
     const { history } = this.props;
 
-    this.props.login(userData).then(success => {
-      if (success){
-        history.push('/kingdom');
-      }
-    });
-   }
+    generalFetch('sessions', {
+      method: 'POST',
+      body: JSON.stringify(userData),
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+    })
+      .then(data => {
+        if (data.error) {
+          return false;
+        } else {
+          localStorage.setItem('TRIBES_TOKEN', data.token);
+          localStorage.setItem('kingdom', data.kingdomId);
+          return true;
+        }
+      })
+      .then(success => {
+        if (success) {
+          history.push('/kingdom/buildings');
+        }
+      });
+  }
```

Comment:
@kkeszthe this should be in a login action, as it was, not just storing these values in the localStorage, but storing the kingdomId redux, since right now you need it